src/rideables/MontageStack.hpp

```diff
@@ -0,0 +1,164 @@
+#ifndef MONTAGESTACK_HPP
+#define MONTAGESTACK_HPP
+
+#include <stdio.h>
+#include <iostream>
+#include <atomic>
+#include <algorithm>
+#include "HarnessUtils.hpp"
+#include "ConcurrentPrimitives.hpp"
+#include "RStack.hpp"
+#include "RCUTracker.hpp"
+#include "CustomTypes.hpp"
+
+using namespace std;
+
+template <typename T>
+class MontageStack : public RStack<T>, Recoverable
+{
+
+private:
+    struct StackNode
+    {
+        T data;
+        StackNode *next;
+        Payload *data;
+
+        MontageStack *ds = nullptr;
+
+        void set_sn(uint64_t s)
+        {
+            assert(payload != nullptr && "payload shouldn't be null");
+            payload->set_unsafe_sn(ds, s);
+        }
+        StackNode(){};
+        StackNode(T v) : next(nullptr), data(pnew<Payload>(v)){};
```

Comment:
pnew Payload passing only one parameter, but `Payload` constructors takes two? This won't compile..